packages/server/modules/workspaces/graph/resolvers/workspaces.ts

```diff
@@ -1,26 +1,101 @@
+  getWorkspaceRolesForUserFactory,
+  upsertWorkspaceFactory,
+  upsertWorkspaceRoleFactory
+} from '@/modules/workspaces/repositories/workspaces'
+import {
+  createWorkspaceFactory,
+  updateWorkspaceFactory
+} from '@/modules/workspaces/services/management'
+import db from '@/db/knex'
+import { getEventBus } from '@/modules/shared/services/eventBus'
+import { getWorkspacesForUserFactory } from '@/modules/workspaces/services/retrieval'
+import { authorizeResolver } from '@/modules/shared'
+import { Roles } from '@speckle/shared'
 
 const { FF_WORKSPACES_MODULE_ENABLED } = getFeatureFlags()
 
 export = FF_WORKSPACES_MODULE_ENABLED
   ? ({
       Query: {
-        workspace: async () => {
-          // Get workspace by id
-          throw new WorkspacesNotYetImplementedError()
+        workspace: async (_parent, args) => {
+          const { id: workspaceId } = args
+
+          // TODO: Use dataloader
+          const workspace = await getWorkspaceFactory({ db })({ workspaceId })
+
+          if (!workspace) {
+            throw new WorkspaceNotFoundError()
+          }
+
+          return workspace
         }
       },
+      Mutation: {
+        workspaceMutations: () => ({})
+      },
       WorkspaceMutations: {
-        create: async () => {
-          throw new WorkspacesNotYetImplementedError()
+        create: async (_parent, args, context) => {
+          const { name, description, logoUrl } = args.input
+
+          const { emit: emitWorkspaceEvent } = getEventBus()
+
+          const upsertWorkspace = upsertWorkspaceFactory({ db })
+          const upsertWorkspaceRole = upsertWorkspaceRoleFactory({ db })
+          // TODO: Integrate with blobstorage
+          const storeBlob = async () => ''
+
+          const createWorkspace = createWorkspaceFactory({
+            upsertWorkspace,
+            upsertWorkspaceRole,
+            emitWorkspaceEvent,
+            storeBlob
+          })
+
+          const workspace = await createWorkspace({
+            userId: context.userId!,
+            workspaceInput: {
+              name,
+              description: description ?? null,
+              logoUrl: logoUrl ?? null
+            }
+          })
+
+          return workspace
         },
         delete: async () => {
           throw new WorkspacesNotYetImplementedError()
         },
-        update: async () => {
-          throw new WorkspacesNotYetImplementedError()
+        update: async (_parent, args, context) => {
+          const { id: workspaceId, ...workspaceInput } = args.input
+
+          await authorizeResolver(context.userId, workspaceId, Roles.Workspace.Admin)
+
+          const { emit: emitWorkspaceEvent } = getEventBus()
+
+          const getWorkspace = getWorkspaceFactory({ db })
+          const upsertWorkspace = upsertWorkspaceFactory({ db })
+          // TODO: Integrate with blobstorage
+          const storeBlob = async () => ''
+
+          const updateWorkspace = updateWorkspaceFactory({
```

Comment:
I like to put those authorizeResolver checks in the service layer to ensure that access rights are checked even if the update is invoked outside of the GQL API